theme/views.py

```diff
@@ -30,11 +30,21 @@ def get_context_data(self, **kwargs):
             except:
                 u = User.objects.get(username=self.request.GET['user'])
 
-        res = BaseResource.objects.filter(user=u)
+        # get all resources the profile user has created
+        resources = BaseResource.objects.filter(user=u)
+
+        # if requesting user is not the profile user, then show only resources that the requesting user has access
+        if self.request.user != u:
+            if self.request.user.is_authenticated():
+                # filter out any resources the requesting user doesn't have view permission
+                resources = [res for res in resources if self.request.user.uaccess.can_view_resource(res)]
+            else:
+                # for anonymous requesting user show only resources that are either public or discoverable
+                resources = [res for res in resources if res.raccess.public or res.raccess.discoverable]
```

Comment:
I am copying the snippet of can_view_resource() below so we have a better idea on how expensive this check is: 
if access_resource.public:
            return True
if UserResourcePrivilege.objects.filter(resource=access_resource, privilege__lte=PrivilegeCodes.VIEW,user=self):
            return True
if GroupResourcePrivilege.objects.filter(resource=access_resource,privilege__lte=PrivilegeCodes.VIEW,             group__members=self):
            return True
It adds a couple of additional filtering to check for view privileges. It may have performance implications if a user has large number of resources, but may not be too bad given that it only filters the resource set for one user.